benchexec/tools/cpachecker.py

```diff
@@ -180,16 +183,21 @@ def option_present(option):
                         f"Unsupported data_model '{data_model}' defined for task '{task}'"
                     )
 
+        if isinstance(task.options, dict) and "witness" in task.options.keys():
+            if not option_present("witness"):
+                options += [f"{prefix}witness", get_unique_witness(task)]
+            else:
+                raise benchexec.tools.template.UnsupportedFeatureException(
+                    "You are passing a witness as both an option and through the task definition. "
+                    "Please remove one of them."
+                )
```

Comment:
Can `get_witness_options` not be used here?

Reply:
Thanks for pointing out this simplification!

Just implemented it